### DISCUSSION TITLE
unbricked ProHD issues - solved (firmware update)

### DISCUSSION THREAD

**Discussion by swilsonnc** [GerryDazoo/Slinger#439] [General]
So I am trying to get this working and am having issues with my ProHD.  It isn't bricked and works flawless if I use slingplayer 1.5 to stream with.  All of this is on my own internal network atm for testing.  If I uncomment VideoSource and set it to 1 I get an error almost evrey time.  If I comment VideoSOurce out it defaults to 0 which is probably wrong it says but it doesn't error.  When I do get video, on the off chance it works, it is very bad and the audio is just a swueal every now and then.  I am using VLC to test.  Any ideas?

Version : 3.08f Running on Windows-10-10.0.19045-SP0 pid= 17604 slingbox_server.exe
Using config file config.ini
Connection Manager Running on port 60000 with 5 max streams using URL slingbox.
BOXES [('sb1', 'ProHD')]
Streamer Running:  5 config.ini ProHD ProHD 60000 8388608
starting up on  port 60000 8388608
Checking for slingbox at ('192.168.1.58', 5001)
('192.168.1.58', 5001) OK
ProHD Using slingbox at  ('192.168.1.58', 5001)
Streamer:  ProHD Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Building page for ProHD
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
12/18/2022, 12:43:45.618   Streaming request from ('192.168.1.13', 57875)

Slinginfo  "Solo/Pro/ProHD" 15 60 192.168.1.58 5001 3000 10
ProHD Starting Stream for  192.168.1.13:57875
Connecting... ('192.168.1.58', 5001) Control
cmd: 0xc6 err: 0xf 56
ProHD Selecting Video Source 1
Traceback (most recent call last):
  File "slingbox_server.py", line 879, in streamer
    client_socket.sendall(OK)
  File "slingbox_server.py", line 495, in start_slingbox_session
    codes = []
IndexError: index out of range
ProHD Badness starting slingbox session  index out of range None
Streamer:  ProHD Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

Config.ini
; This a sample of a unified config.ini for three slingboxes

[SLINGBOXES]
sb1=ProHD

[ProHD]
sbtype="Solo/Pro/ProHD"
password=xxxxxxxxxx     
ipaddress=192.168.1.58
;finderid=BAD95DAED609364C8204F2BC5FBDC363
port=5001
Resolution=15
FrameRate=60
VideoBandwidth=4000
VideoSmoothness=63
AudioBitRate=32
VideoSource=1
RemoteLock=yes
;StartChannel=155
Remote=remote.txt
;Code=1

[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number.
port=60000
maxremotestreams=5
URLbase=slingbox
enableremote=yes

**Comment by BobC1954** [comment]
Video source 0 is correct for the internal tuner on ProHD

**Comment by swilsonnc** [comment]
Yes, that is correct but I am not using the internal tuner.  I am using Composite and setting it to 1 but it errors usually.

**Reply by GerryDazoo** [reply]
Please post these "errors" you are referring to. If it's a program bug I'd like to fix it,

**Reply by swilsonnc** [reply]
It was actually several errors but don’t have them all saved.  It seemed a lot of it was caused by the remote code retrieval from the SB with the old firmware.  What I posted is about all I have saved since it is working now.  Working much better than it ever did with Slingplayer.

**Comment by swilsonnc** [comment]
Well, I found my issue.  Apparentely I have had this ProHD for too long and never updated the firmware in it.  It was on 1.1.250 and having issues.  I update it using Slingplayer 1.5 to 1.5.170 (last firmware before the 2.0 series) and it works great.  

Maybe others on here could check and post their firmware for their ProHD's so we know what works and what doesn't?  Get the firmware version in Information tab of the Slingbox in Slingplayer 1.5.  Anyone using any 2.0 versions successfully?

**Reply by GerryDazoo** [reply]
The latest firmware is SbPbProHdFwV2_01_420_HwV1_0_0Current.bin. Why would you think it will not work?

**Reply by swilsonnc** [reply]
While I was testing I wanted to see if any of the older version 1 firmware would work.  I updated to 2.1.420 shortly after finding out that was the problem.
